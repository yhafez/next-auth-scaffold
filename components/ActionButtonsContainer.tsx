// Path: ./components/ActionButtonsContainer.tsx
import { ReactNode } from 'react'
import { Box } from '@mui/material'

export interface ActionButtonsContainerProps {
	name: string
	children: ReactNode
}

export default function ActionButtonsContainer({ name, children }: ActionButtonsContainerProps) {
	return (
		<Box
			id={`${name}-buttons-container`}
			sx={{
				display: 'flex',
				flexDirection: 'row',
				justifyContent: 'space-around',
				alignItems: 'center',
				gap: 2,
				margin: '8px 4px',
				width: '80%',
			}}
		>
			{children}
		</Box>
	)
}
