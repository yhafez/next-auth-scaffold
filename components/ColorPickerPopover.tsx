// Path: ./components/ColorPickerPopover.tsx

import { useState, useCallback } from 'react'
import { Popover, PopoverOrigin } from '@mui/material'
import { useBoundStore } from '../store'
import { ChromePicker } from 'react-color'

export interface ColorPickerPopoverProps {
	name: string
	open: boolean
	anchorEl: HTMLButtonElement | null
	handleClose: () => void
	handleColorChange: (color: string) => void
	anchorOrigin?: PopoverOrigin
	transformOrigin?: PopoverOrigin
}

const ColorPickerPopover = ({
	name,
	open,
	anchorEl,
	handleClose,
	handleColorChange,
	anchorOrigin,
	transformOrigin,
}: ColorPickerPopoverProps) => {
	const { customPalette } = useBoundStore()

	const [color, setColor] = useState(customPalette.primary.main)

	const handleChange = useCallback(
		(color: any) => {
			setColor(color.hex)
			handleColorChange(color.hex)
		},
		[handleColorChange],
	)

	return (
		<Popover
			id={`${name}-color-picker-popover`}
			open={open}
			anchorEl={anchorEl}
			onClose={handleClose}
			anchorOrigin={anchorOrigin}
			transformOrigin={transformOrigin}
			sx={{
				cursor: 'pointer',
				'& .MuiPopover-paper': {
					padding: 2,
					backgroundColor: 'background.paper',
					color: 'text.primary',
				},
			}}
		>
			<ChromePicker color={color} onChange={color => handleChange(color)} disableAlpha />
		</Popover>
	)
}

export default ColorPickerPopover
