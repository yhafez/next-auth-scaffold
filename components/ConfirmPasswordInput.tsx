// Path: ./components/ConfirmPasswordInput.tsx
import { Dispatch, SetStateAction, KeyboardEvent } from 'react'

import PasswordInput from './PasswordInput'

export interface ConfirmPasswordInputProps {
	name: string
	password: string
	setPassword: Dispatch<SetStateAction<string>>
	confirmPassword: string
	setConfirmPassword: Dispatch<SetStateAction<string>>
	disabled?: boolean
	handleEnter?: (e: KeyboardEvent) => void
}

export default function ConfirmPasswordInput({
	name,
	password,
	setPassword,
	confirmPassword,
	setConfirmPassword,
	disabled = false,
	handleEnter,
}: ConfirmPasswordInputProps) {
	return (
		<>
			<PasswordInput
				name={name}
				value={password}
				setValue={setPassword}
				disabled={disabled}
				handleEnter={handleEnter}
			/>
			<PasswordInput
				name={`${name}`}
				value={confirmPassword}
				setValue={setConfirmPassword}
				disabled={disabled}
				isConfirmPassword={true}
				handleEnter={handleEnter}
			/>
		</>
	)
}
