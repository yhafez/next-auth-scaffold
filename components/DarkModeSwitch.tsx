// Path: ./components/DarkModeSwitch.tsx
import { useEffect } from 'react'
import { Box, Switch, Typography } from '@mui/material'
import WbSunnyIcon from '@mui/icons-material/WbSunny'
import Brightness2Icon from '@mui/icons-material/Brightness2'

import { useBoundStore } from '../store'

export interface DarkModeSwitchProps {
	name: string
}

export default function DarkModeSwitch({ name }: DarkModeSwitchProps) {
	const { darkMode, setDarkMode } = useBoundStore()

	const handleDarkMode = () => {
		localStorage.setItem('darkMode', JSON.stringify(!darkMode))
		setDarkMode(!darkMode)
	}

	useEffect(() => {
		const darkMode = JSON.parse(localStorage.getItem('darkMode') || 'false')
		setDarkMode(darkMode)
	}, [darkMode, setDarkMode])

	return (
		<Box
			id={`${name}-dark-mode-switch-container`}
			sx={{
				display: 'flex',
				flexDirection: 'column',
				justifyContent: 'center',
				alignItems: 'center',
				gap: 1,
				mt: 1,
			}}
		>
			<label htmlFor={`${name}-dark-mode-switch`}>
				<Switch
					id={`${name}-dark-mode-switch`}
					checked={darkMode}
					onChange={handleDarkMode}
					sx={{
						'& .MuiSwitch-switchBase': {
							color: 'primary.contrastText',
							'&.Mui-checked': {
								color: 'primary.contrastText',
							},
							'&.Mui-checked + .MuiSwitch-track': {
								backgroundColor: 'secondary.main',
							},
						},
					}}
					checkedIcon={<Brightness2Icon id={`${name}-dark-mode-switch-icon-checked`} />}
					icon={<WbSunnyIcon id={`${name}-dark-mode-switch-icon-unchecked`} />}
				/>
				<Typography
					id={`${name}-dark-mode-switch-label-${darkMode ? 'dark' : 'light'}`}
					variant="body2"
					sx={{ fontWeight: 500, mb: 1, color: 'primary.contrastText' }}
				>
					{darkMode ? 'Dark Mode' : 'Light Mode'}
				</Typography>
			</label>
		</Box>
	)
}
