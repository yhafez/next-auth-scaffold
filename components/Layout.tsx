// Path: components/Layout.tsx

import { ReactNode } from 'react'
import { AppBar, Box, Toolbar, Typography, useMediaQuery } from '@mui/material'

import DarkModeSwitch from './DarkModeSwitch'
import ColorPickerIcon from './ColorPickerIcon'

import { useBoundStore } from '../store'

export interface LayoutProps {
	children: ReactNode
	name: string
	drawerWidth?: number
}

export function Layout({ children, name, drawerWidth }: LayoutProps) {
	const { darkMode, customPalette, theme } = useBoundStore()
	const isMobile = useMediaQuery(theme.breakpoints.down('sm'))

	return (
		<Box
			id={`${name}-layout-container`}
			sx={{
				display: 'flex',
				'& .MuiDrawer-paper': {
					width: drawerWidth,
					boxSizing: 'border-box',
				},
			}}
		>
			<AppBar
				id={`${name}-app-bar`}
				position="fixed"
				sx={{
					backgroundColor: darkMode ? 'primary.dark' : 'primary.light',
					color: customPalette.primary.contrastText,
					transition: theme.transitions.create(['width', 'margin'], {
						easing: theme.transitions.easing.sharp,
						duration: theme.transitions.duration.leavingScreen,
					}),
					width: `${isMobile ? '100%' : `calc(100% - ${drawerWidth}px)`}`,
					ml: `${isMobile ? 0 : `${drawerWidth}px`}`,
					height: '64px',
				}}
			>
				<Toolbar
					id={`${name}-toolbar`}
					sx={{
						display: 'flex',
						justifyContent: 'space-between',
						alignItems: 'center',
					}}
				>
					<Typography
						id={`${name}-toolbar-title`}
						variant="h6"
						noWrap
						component="h1"
						sx={{
							flexGrow: 1,
							color: 'primary.contrastText',
						}}
					>
						Next Auth Scaffold
					</Typography>
					<ColorPickerIcon name={`${name}-toolbar`} />
					<DarkModeSwitch name={`${name}-toolbar`} />
				</Toolbar>
			</AppBar>
			<Box
				id={`${name}-layout-content`}
				sx={{
					flexGrow: 1,
					padding: theme.spacing(3),
					transition: theme.transitions.create(['width', 'margin'], {
						easing: theme.transitions.easing.sharp,
						duration: theme.transitions.duration.leavingScreen,
					}),
					marginLeft: drawerWidth,
					width: `${isMobile ? '100%' : `calc(100% - ${drawerWidth}px)`}`,
					ml: `${isMobile ? 0 : `${drawerWidth}px`}`,
					mt: '64px',
					display: 'flex',
					flexDirection: 'column',
					justifyContent: 'center',
					alignItems: 'center',
					height: 'calc(100vh - 64px)',
					backgroundColor: darkMode ? 'grey.900' : 'grey.100',
				}}
			>
				{children}
			</Box>
		</Box>
	)
}
