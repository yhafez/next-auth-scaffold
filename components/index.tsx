// Path: ./components/index.tsx
import Layout from './Layout'

import DashboardDrawer from './Drawers/DashboardDrawer'
import SettingsDrawer from './Drawers/SettingsDrawer'

import ColorPickerPopover from './Popovers/ColorPickerPopover'
import DrawerSignOutPopover from './Popovers/DrawerSignOutPopover'

import ActionButtonsContainer from './ActionButtonsContainer'
import ColorPickerIcon from './ColorPickerIcon'
import ConfirmPasswordInput from './ConfirmPasswordInput'
import DarkModeSwitch from './DarkModeSwitch'
import Drawer from './Drawer'
import DrawerButton from './DrawerButton'
import EditableTextField from './EditableTextField'
import EmailInput from './EmailInput'
import Modal from './Modal'
import ModalButton from './ModalButton'
import ModalNote from './ModalNote'
import NavigationButton from './NavigationButton'
import PasswordInput from './PasswordInput'
import SubmitButton from './SubmitButton'
import VisibilityIconButton from './VisibilityIconButton'

export {
	Layout,
	DashboardDrawer,
	SettingsDrawer,
	ColorPickerPopover,
	DrawerSignOutPopover,
	ActionButtonsContainer,
	ColorPickerIcon,
	ConfirmPasswordInput,
	DarkModeSwitch,
	Drawer,
	DrawerButton,
	EditableTextField,
	EmailInput,
	Modal,
	ModalButton,
	ModalNote,
	NavigationButton,
	PasswordInput,
	SubmitButton,
	VisibilityIconButton,
}
