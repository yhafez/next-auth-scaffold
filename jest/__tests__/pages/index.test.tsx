// Path: ./jest/__tests__/pages/index.test.tsx
import { render, act, screen, waitFor } from '@testing-library/react'
import { axe } from 'jest-axe'
import { ThemeProvider } from '@mui/material/styles'

import Dashboard from '../../../pages'
import { Layout } from '../../../components/Layout'
import { defaultTheme } from '../../../theme'
import { SessionProvider } from 'next-auth/react'

describe('Dashboard', () => {
	it('Should have no accessibility violations', async () => {
		act(() =>
			render(
				<ThemeProvider theme={defaultTheme}>
					<SessionProvider
						session={{
							user: {
								name: 'Test User',
								email: 'test@example.com',
								image: 'https://example.com/image.png',
							},
							expires: '1',
						}}
					>
						<Layout name="test">
							<main>
								<Dashboard />
							</main>
						</Layout>
					</SessionProvider>
				</ThemeProvider>,
			),
		)

		waitFor(async () => expect(await axe(screen.getByRole('main'))).toHaveNoViolations())
	})

	it('Should match snapshot', () => {
		const { container } = render(
			<ThemeProvider theme={defaultTheme}>
				<SessionProvider
					session={{
						user: {
							name: 'Test User',
							email: 'test@example.com',
							image: 'https://example.com/image.png',
						},
						expires: '1',
					}}
				>
					<Layout name="test">
						<Dashboard />
					</Layout>
				</SessionProvider>
			</ThemeProvider>,
		)

		expect(container).toMatchSnapshot()
	})
})
