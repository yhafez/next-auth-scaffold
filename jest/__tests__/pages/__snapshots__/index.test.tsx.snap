// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-86scw3"
    id="test-layout-container"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-7t767d-MuiPaper-root-MuiAppBar-root"
      id="test-app-bar"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1mg7r1q-MuiToolbar-root"
        id="test-toolbar"
      >
        <h1
          class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-11cao63-MuiTypography-root"
          id="test-toolbar-title"
        >
          Next Auth Scaffold
        </h1>
        <div
          class="MuiBox-root css-16hvbxa"
          id="test-toolbar-color-picker-container"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-xfhdu6-MuiButtonBase-root-MuiIconButton-root"
            id="test-toolbar-color-picker-button"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1kqeszy-MuiSvgIcon-root"
              data-testid="ColorLensIcon"
              focusable="false"
              id="test-toolbar-color-picker-icon"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9c.83 0 1.5-.67 1.5-1.5 0-.39-.15-.74-.39-1.01-.23-.26-.38-.61-.38-.99 0-.83.67-1.5 1.5-1.5H16c2.76 0 5-2.24 5-5 0-4.42-4.03-8-9-8zm-5.5 9c-.83 0-1.5-.67-1.5-1.5S5.67 9 6.5 9 8 9.67 8 10.5 7.33 12 6.5 12zm3-4C8.67 8 8 7.33 8 6.5S8.67 5 9.5 5s1.5.67 1.5 1.5S10.33 8 9.5 8zm5 0c-.83 0-1.5-.67-1.5-1.5S13.67 5 14.5 5s1.5.67 1.5 1.5S15.33 8 14.5 8zm3 4c-.83 0-1.5-.67-1.5-1.5S16.67 9 17.5 9s1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
              />
            </svg>
            <p
              class="MuiTypography-root MuiTypography-body2 css-1uk2y8w-MuiTypography-root"
              id="test-toolbar-color-picker-label"
            >
              Theme Color
            </p>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiBox-root css-4o796m"
          id="test-toolbar-dark-mode-switch-container"
        >
          <label
            for="test-toolbar-dark-mode-switch"
          >
            <span
              class="MuiSwitch-root MuiSwitch-sizeMedium css-1dkykqx-MuiSwitch-root"
            >
              <span
                class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-28iv8m-MuiButtonBase-root-MuiSwitch-switchBase"
              >
                <input
                  class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                  id="test-toolbar-dark-mode-switch"
                  type="checkbox"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="WbSunnyIcon"
                  focusable="false"
                  id="test-toolbar-dark-mode-switch-icon-unchecked"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m6.76 4.84-1.8-1.79-1.41 1.41 1.79 1.79 1.42-1.41zM4 10.5H1v2h3v-2zm9-9.95h-2V3.5h2V.55zm7.45 3.91-1.41-1.41-1.79 1.79 1.41 1.41 1.79-1.79zm-3.21 13.7 1.79 1.8 1.41-1.41-1.8-1.79-1.4 1.4zM20 10.5v2h3v-2h-3zm-8-5c-3.31 0-6 2.69-6 6s2.69 6 6 6 6-2.69 6-6-2.69-6-6-6zm-1 16.95h2V19.5h-2v2.95zm-7.45-3.91 1.41 1.41 1.79-1.8-1.41-1.41-1.79 1.8z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
              />
            </span>
            <p
              class="MuiTypography-root MuiTypography-body2 css-1e5jxpk-MuiTypography-root"
              id="test-toolbar-dark-mode-switch-label-light"
            >
              Light Mode
            </p>
          </label>
        </div>
      </div>
    </header>
    <div
      class="MuiBox-root css-eohel5"
      id="test-layout-content"
    >
      <div
        class="MuiBox-root css-3j596c"
        id="dashboard"
      >
        <div
          class="MuiDrawer-root MuiDrawer-docked css-1kr353o-MuiDrawer-docked"
          id="drawer"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-12i7wg6-MuiPaper-root-MuiDrawer-paper"
          >
            <div
              class="MuiBox-root css-1h6p0qj"
              id="drawer-header"
            >
              <div
                class="MuiBox-root css-1vwnqvu"
                id="drawer-header-user"
              >
                <label
                  for="avatar"
                  style="width: 100%;"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation css-k9vxts-MuiButtonBase-root-MuiButton-root"
                    id="drawer-user-avatar"
                    tabindex="0"
                    type="button"
                  >
                    <div
                      class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-9d7ivx-MuiAvatar-root"
                      id="drawer-user-avatar-image"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
                        data-testid="PersonIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                        />
                      </svg>
                    </div>
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-10xoo56-MuiTypography-root"
                      id="drawer-user-avatar-name-placeholder"
                    >
                      User
                    </p>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </label>
              </div>
              <div
                class="MuiBox-root css-167ywge"
                id="drawer-header-actions"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-hmb9pg-MuiButtonBase-root-MuiIconButton-root"
                  id="menu-drawer-button"
                  tabindex="0"
                  type="button"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-zlcmfu-MuiTypography-root"
                    id="menu-drawer-button-label"
                  >
                    Menu
                  </p>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="MenuIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-hmb9pg-MuiButtonBase-root-MuiIconButton-root"
                  id="profile-drawer-button"
                  tabindex="0"
                  type="button"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-zlcmfu-MuiTypography-root"
                    id="profile-drawer-button-label"
                  >
                    Profile
                  </p>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="AccountCircleIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6zm0 14c-2.03 0-4.43-.82-6.14-2.88C7.55 15.8 9.68 15 12 15s4.45.8 6.14 2.12C16.43 19.18 14.03 20 12 20z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-hmb9pg-MuiButtonBase-root-MuiIconButton-root"
                  id="settings-drawer-button"
                  tabindex="0"
                  type="button"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-zlcmfu-MuiTypography-root"
                    id="settings-drawer-button-label"
                  >
                    Settings
                  </p>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="SettingsIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
            <div
              class="MuiBox-root css-19gca34"
              id="drawer-footer"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-hmb9pg-MuiButtonBase-root-MuiIconButton-root"
                id="color-picker-drawer-button"
                tabindex="0"
                type="button"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-zlcmfu-MuiTypography-root"
                  id="color-picker-drawer-button-label"
                >
                  Theme Color
                </p>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="PaletteIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.49 2 2 6.49 2 12s4.49 10 10 10c1.38 0 2.5-1.12 2.5-2.5 0-.61-.23-1.2-.64-1.67-.08-.1-.13-.21-.13-.33 0-.28.22-.5.5-.5H16c3.31 0 6-2.69 6-6 0-4.96-4.49-9-10-9zm5.5 11c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zm-3-4c-.83 0-1.5-.67-1.5-1.5S13.67 6 14.5 6s1.5.67 1.5 1.5S15.33 9 14.5 9zM5 11.5c0-.83.67-1.5 1.5-1.5s1.5.67 1.5 1.5S7.33 13 6.5 13 5 12.33 5 11.5zm6-4c0 .83-.67 1.5-1.5 1.5S8 8.33 8 7.5 8.67 6 9.5 6s1.5.67 1.5 1.5z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-hmb9pg-MuiButtonBase-root-MuiIconButton-root"
                id="dark-mode-drawer-button"
                tabindex="0"
                type="button"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-zlcmfu-MuiTypography-root"
                  id="dark-mode-drawer-button-label"
                >
                  Dark Mode
                </p>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="Brightness4Icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zM12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6c3.31 0 6 2.69 6 6s-2.69 6-6 6z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-hmb9pg-MuiButtonBase-root-MuiIconButton-root"
                id="logout-drawer-button"
                tabindex="0"
                type="button"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-zlcmfu-MuiTypography-root"
                  id="logout-drawer-button-label"
                >
                  Logout
                </p>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ExitToAppIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10.09 15.59 11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-oqgvqi"
          id="dashboard-layout-container"
        >
          <header
            class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-vthed8-MuiPaper-root-MuiAppBar-root"
            id="dashboard-app-bar"
          >
            <div
              class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1mg7r1q-MuiToolbar-root"
              id="dashboard-toolbar"
            >
              <h1
                class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-11cao63-MuiTypography-root"
                id="dashboard-toolbar-title"
              >
                Next Auth Scaffold
              </h1>
              <div
                class="MuiBox-root css-16hvbxa"
                id="dashboard-toolbar-color-picker-container"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-xfhdu6-MuiButtonBase-root-MuiIconButton-root"
                  id="dashboard-toolbar-color-picker-button"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1kqeszy-MuiSvgIcon-root"
                    data-testid="ColorLensIcon"
                    focusable="false"
                    id="dashboard-toolbar-color-picker-icon"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9c.83 0 1.5-.67 1.5-1.5 0-.39-.15-.74-.39-1.01-.23-.26-.38-.61-.38-.99 0-.83.67-1.5 1.5-1.5H16c2.76 0 5-2.24 5-5 0-4.42-4.03-8-9-8zm-5.5 9c-.83 0-1.5-.67-1.5-1.5S5.67 9 6.5 9 8 9.67 8 10.5 7.33 12 6.5 12zm3-4C8.67 8 8 7.33 8 6.5S8.67 5 9.5 5s1.5.67 1.5 1.5S10.33 8 9.5 8zm5 0c-.83 0-1.5-.67-1.5-1.5S13.67 5 14.5 5s1.5.67 1.5 1.5S15.33 8 14.5 8zm3 4c-.83 0-1.5-.67-1.5-1.5S16.67 9 17.5 9s1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
                    />
                  </svg>
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-1uk2y8w-MuiTypography-root"
                    id="dashboard-toolbar-color-picker-label"
                  >
                    Theme Color
                  </p>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <div
                class="MuiBox-root css-4o796m"
                id="dashboard-toolbar-dark-mode-switch-container"
              >
                <label
                  for="dashboard-toolbar-dark-mode-switch"
                >
                  <span
                    class="MuiSwitch-root MuiSwitch-sizeMedium css-1dkykqx-MuiSwitch-root"
                  >
                    <span
                      class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-28iv8m-MuiButtonBase-root-MuiSwitch-switchBase"
                    >
                      <input
                        class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                        id="dashboard-toolbar-dark-mode-switch"
                        type="checkbox"
                      />
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="WbSunnyIcon"
                        focusable="false"
                        id="dashboard-toolbar-dark-mode-switch-icon-unchecked"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="m6.76 4.84-1.8-1.79-1.41 1.41 1.79 1.79 1.42-1.41zM4 10.5H1v2h3v-2zm9-9.95h-2V3.5h2V.55zm7.45 3.91-1.41-1.41-1.79 1.79 1.41 1.41 1.79-1.79zm-3.21 13.7 1.79 1.8 1.41-1.41-1.8-1.79-1.4 1.4zM20 10.5v2h3v-2h-3zm-8-5c-3.31 0-6 2.69-6 6s2.69 6 6 6 6-2.69 6-6-2.69-6-6-6zm-1 16.95h2V19.5h-2v2.95zm-7.45-3.91 1.41 1.41 1.79-1.8-1.41-1.41-1.79 1.8z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                    />
                  </span>
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-1e5jxpk-MuiTypography-root"
                    id="dashboard-toolbar-dark-mode-switch-label-light"
                  >
                    Light Mode
                  </p>
                </label>
              </div>
            </div>
          </header>
          <div
            class="MuiBox-root css-40v626"
            id="dashboard-layout-content"
          >
            <div
              class="MuiBox-root css-jv7tdc"
              id="dashboard-content"
            >
              <h2
                class="MuiTypography-root MuiTypography-h2 css-137s1gz-MuiTypography-root"
                id="dashboard-content-title"
              >
                Dashboard
              </h2>
              <p
                class="MuiTypography-root MuiTypography-body1 css-mqfe5n-MuiTypography-root"
                id="dashboard-content-subtitle"
              >
                Welcome 
                !
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
